Adaptors/MongoDB/tests/MongoDatabaseProvider.cs

```diff
@@ -0,0 +1,125 @@
+                                               {
+                                                 {
+                                                   $"{Components.SettingSection}:{nameof(Components.TableStorage)}", "ArmoniK.Adapters.MongoDB.TableStorage"
+                                                 },
+                                                 {
+                                                   $"{Components.SettingSection}:{nameof(Components.ObjectStorage)}", "ArmoniK.Adapters.MongoDB.ObjectStorage"
+                                                 },
+                                                 {
+                                                   $"{Components.SettingSection}:{nameof(Components.AuthenticationStorage)}",
+                                                   "ArmoniK.Adapters.MongoDB.AuthenticationTable"
+                                                 },
+                                                 {
+                                                   $"{Options.MongoDB.SettingSection}:{nameof(Options.MongoDB.DatabaseName)}", DatabaseName
+                                                 },
+                                                 {
+                                                   $"{Options.MongoDB.SettingSection}:{nameof(Options.MongoDB.TableStorage)}:{nameof(Options.MongoDB.TableStorage.PollingDelayMax)}",
+                                                   "00:00:10"
+                                                 },
+                                                 {
+                                                   $"{Options.MongoDB.SettingSection}:{nameof(Options.MongoDB.ObjectStorage)}:{nameof(Options.MongoDB.ObjectStorage.ChunkSize)}",
+                                                   "140000"
+                                                 },
+                                               };
+
+    var configuration = new ConfigurationManager();
+    configuration.AddInMemoryCollection(minimalConfig);
+
+    var serviceCollection = new ServiceCollection();
+    serviceCollection.AddMongoStorages(configuration,
+                                       logger);
+    serviceCollection.AddClientSubmitterAuthenticationStorage(configuration,
+                                                              logger);
+    serviceCollection.AddSingleton(ActivitySource);
+    serviceCollection.AddTransient<IMongoClient>(_ => client);
+
+    serviceCollection.AddLogging();
+    serviceConfigurator?.Invoke(serviceCollection);
+
+    provider_ = serviceCollection.BuildServiceProvider(new ServiceProviderOptions
+                                                       {
+                                                         ValidateOnBuild = true,
+                                                       });
+  }
+
+  public void Dispose()
+  {
+    provider_.Dispose();
+    runner_?.Dispose();
+  }
+
+  public IServiceProvider GetServiceProvider()
+    => provider_;
+}
```

Comment:
I think we could move this to the Test Helpers dir, the same code is used a least in the submitter tests

Reply:
I've seen it in 2 other classes but the integration in those tests is more complicated than for those.